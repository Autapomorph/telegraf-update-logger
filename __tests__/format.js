const format = require('../lib/format');
const testUpdates = require('./testUpdates.json');

describe('telegraf-update-logger/lib/format', () => {
  describe('export default function format', () => {
    describe('it works when message is', () => {
      Object.keys(testUpdates).forEach(testName => {
        it(testName, () => {
          // given:
          const update = testUpdates[testName];
          // when:
          const formattedUpdate = format(update);
          // then:
          expect(formattedUpdate).toMatchSnapshot();
        });
      });
    });
  });
});
