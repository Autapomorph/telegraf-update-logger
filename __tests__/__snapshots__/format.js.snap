// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`telegraf-update-logger/lib/format export default function format it works when message is audio 1`] = `"[16] Foo Bar: document"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is audio with caption 1`] = `"[16] Foo Bar: document, caption"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is channel post 1`] = `"[5] Test Channel:: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is channel post with author signature 1`] = `"[6] Test Channel: Foo Bar: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is contact 1`] = `"[22] Foo Bar: contact of Some One"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is document 1`] = `"[17] Foo Bar: document"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is edited 1`] = `"[9] Foo Bar (edited): text"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is edited channel post 1`] = `"[5] Test Channel: (edited): text"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is emoji sticker 1`] = `"[13] Foo Bar: ðŸ˜Ž   sticker"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is forwarded from channel 1`] = `"[8] Foo Bar fwd[Test Channel]: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is forwarded from user 1`] = `"[7] Foo Bar fwd[Baz]: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is from user 1`] = `"[1] Foo Bar: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is from user without a last name 1`] = `"[2] Baz: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is in group 1`] = `"[3] Test Group: Foo Bar: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is in super group 1`] = `"[4] Test Supergroup: Foo Bar: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is left chat member 1`] = `"[25] Test Group: Foo Bar: removed Baz"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is location 1`] = `"[21] Foo Bar: location on 50.450091 30.523415"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is new chat members 1`] = `"[24] Test Group: Foo Bar: added Baz"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is new chat photo 1`] = `"[27] Test Group: Foo Bar: changed chat photo"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is new chat title 1`] = `"[26] New Group: Foo Bar: changed chat title"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is photo 1`] = `"[14] Foo Bar: photo"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is photo with caption 1`] = `"[15] Foo Bar: photo, caption"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is reply 1`] = `"[9] Foo Bar re[1]: message"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is sticker 1`] = `"[12] Foo Bar: sticker"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is video 1`] = `"[18] Foo Bar: video"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is video note 1`] = `"[23] Foo Bar: video_note"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is video with caption 1`] = `"[19] Foo Bar: video, caption"`;

exports[`telegraf-update-logger/lib/format export default function format it works when message is voice 1`] = `"[20] Foo Bar: voice"`;
